src/v3/recordingComputeUtils.ts

```diff
@@ -72,14 +76,126 @@ const markedInteractive = <RelationSchemasT>(
   spanAndAnnotation: SpanAndAnnotation<RelationSchemasT>,
 ) => spanAndAnnotation.annotation.markedPageInteractive
 
+/**
+ * Helper function to create a matcher function from a definition
+ */
+function createMatcher<
```

Comment:
isn't this function basically the same as `ensureMatcherFn`, just with the added functionality of handling special strings (like 'operation-start')? Maybe we can just create `ensureMatcherFnOrSpecialToken` that wraps `ensureMatcherFn`?